prime-router/src/main/kotlin/Mappers.kt

```diff
@@ -406,11 +406,35 @@ class SplitMapper : Mapper {
     override fun apply(element: Element, args: List<String>, values: List<ElementAndValue>): String? {
         if (values.isEmpty()) return null
         val value = values.firstOrNull()?.value ?: ""
-        val splitElements = value.split(" ")
+        val delimiter = if (args.count() > 2) {
+            args[2]
+        } else {
+            " "
+        }
+        val splitElements = value.split(delimiter)
+        val index = args[1].toInt()
+        if (index > splitElements.count()) error("Index $index exceeds count of split values in $value")
+
+        return splitElements[index].trim()
+    }
+}
+
+class SplitByCommaMapper : Mapper {
+    override val name = "splitByComma"
+
+    override fun valueNames(element: Element, args: List<String>): List<String> {
+        return listOf(args[0])
+    }
+
+    override fun apply(element: Element, args: List<String>, values: List<ElementAndValue>): String? {
+        if (values.isEmpty()) return null
+        val value = values.firstOrNull()?.value ?: ""
+        val delimiter = ","
+        val splitElements = value.split(delimiter)
         val index = args[1].toInt()
         if (index > splitElements.count()) error("Index $index exceeds count of split values in $value")
 
-        return splitElements[index]
+        return splitElements[index].trim()
```

Comment:
Fun fact our parsing for mappers doesn't like trying to pass through commas as arguments. I tried many different ways including UTF encoding and writing them in hex. Nope.

Reply:
Not surprised - we're trying to use regex's to parse our fake mapper "code", whose complexity is teetering on the edge of needing a real recursive descent parser.